src/library_pthread.js

```diff
@@ -814,13 +832,15 @@ var LibraryPThread = {
       // new thread, the thread creation must be deferred to the main JS thread.
       threadParams.cmd = 'spawnThread';
       postMessage(threadParams, transferList);
-    } else {
-      // We are the main thread, so we have the pthread warmup pool in this
-      // thread and can fire off JS thread creation directly ourselves.
-      spawnThread(threadParams);
+      // We don't care about actual exit code, because spawnThread
+      // only returns non-0 as part of the PTHREAD_POOL_SIZE_STRICT check, which
+      // doesn't apply to the PROXY_TO_PTHREAD mode we're in anyway.
```

Comment:
That's what I originally thought, and added a test to verify, but apparently not - it's taking another branch above:

https://github.com/emscripten-core/emscripten/blob/cdcf0ae5fcf6c9a1d52fa356f2ade821c39aa67e/src/library_pthread.js#L727

Reply:
If this code is really only designed for `PROXY_TO_PTHREAD` then it should be enclodes in `#ifdef PROXY_TO_PTHREAD` .    It seems like this postMessage version is used whenever there is a transferList (i.e. whenever there is offsetscreen canvas). 

Wouldn't it make more sense to say "only returns non-0 as part of the PTHREAD_POOL_SIZE_STRICT check, which only runs on the main thread and not when ENVIRONMENT_IS_PTHREAD.".

Reply:
Perhaps it would even better to say something like:  "When using postMessage to spawn a thread we cannot detect failures synchronously so we are forced to assume success and return 0 here".

Reply:
> If this code is really only designed for `PROXY_TO_PTHREAD` then it should be enclodes in `#ifdef PROXY_TO_PTHREAD` . It seems like this postMessage version is used whenever there is a transferList (i.e. whenever there is offsetscreen canvas).

Ah maybe... I'm having hard time understanding what this covers / tries to do tbh, but I suppose you're right that it applies to OffscreenCanvas usages as well.

Reply:
> "When using postMessage to spawn a thread we cannot detect failures synchronously so we are forced to assume success and return 0 here".

FWIW this is not entirely true - we could instead use a shared atomic indicating number of free threads, and synchronously check the value even here. But I don't know if it's worth fixing, especially since I can't yet come up with an example where it'd fail...

Reply:
But is true today, right?   How about:  "We have no mechanism to detect failure synchronously today so we have to return 0".

Reply:
Reworded & updated.